apps/docs/content/guides/auth/third-party/clerk.mdx

```diff
@@ -0,0 +1,142 @@
+  accessToken: () => {
+    return Clerk.session?.getToken()
+  },
+})
+```
+
+</TabPanel>
+
+<TabPanel id="dart" label="Flutter">
+
+```dart
+import 'package:clerk_flutter/clerk_flutter.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+...
+
+await Supabase.initialize(
+  url: 'SUPABASE_URL',
+  anonKey: 'SUPABASE_ANON_KEY',
+  accessToken: () async {
+    final token = await ClerkAuth.of(context).sessionToken();
+    return token.jwt;
+  },
+);
+```
+
+</TabPanel>
+
+<TabPanel id="swift" label="Swift (iOS)">
+
+```swift
+import Clerk
+import Supabase
+
+let supabase = SupabaseClient(
+  supabaseURL: URL(string: "https://project-ref.supabase.io")!,
+  supabaseKey: "supabase.anon.key",
+  options: SupabaseClientOptions(
+    auth: SupabaseClientOptions.AuthOptions(
+      accessToken: {
+        try await Clerk.shared.session?.getToken()?.jwt
+      }
+    )
+  )
+)
+```
+
+</TabPanel>
+
+</Tabs>
+
+## Using RLS policies
+
+Once you've configured the Supabase client library to use Clerk session tokens, you can use RLS policies to secure access to your project's database, Storage objects and Realtime channels.
+
+The recommended way to design RLS policies with Clerk is to use claims present in your Clerk session token to allow or reject access to your project's data. Check [Clerk's docs](https://clerk.com/docs/backend-requests/resources/session-tokens) on the available JWT claims and their values.
+
+### Example: Check user organization role
+
+```sql
+create policy "Only organization admins can insert in table"
+  on table_name
+  to authenticated
+  with check (
+    (auth.jwt()->>'org_role' = 'org:admin')
+      and
+    (organization_id = auth.jwt()->>'org_id')
+  );
+```
+
+This RLS policy checks that the newly inserted row in the table has the user's declared organization ID (from the `org_id` session token claim) in the `organization_id` column. Additionally it ensures that they're an `org:admin`.
+
+This way only organization admins can add rows to the table, for organizations they're a member of.
+
+### Example: Check user has passed second factor verification
+
+```sql
+create policy "Only users that have passed second factor verification can read from table"
+  on table_name
+  as restrictive
+  to authenticated
+  using (
+    (auth.jwt()->'fva'->>1 != '-1')
+  );
+```
+
+This example uses a restrictive RLS policy checks that the second factor verification age element in the `fva` claim is not `'-1'` indicating the user has passed through second factor verification.
+
+## Deprecated integration with JWT templates
+
+As of 1st April 2025 the previously available [Clerk Integration with Supabase](https://supabase.com/partners/integrations/clerk) is considered deprecated and is no longer recommended for use. Additionally all projects using this integration will not be charged for Third-Party Monthly Active Users (TP-MAU) until at least 1st January 2026.
```

Comment:
Just wanna clarify that "this integration" here is the old one not the new one?

Reply:
Old one yes, I'll rephrase.